src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs

```diff
@@ -116,11 +116,9 @@ public async Task<int> ExecuteCommand(PackageReferenceArgs packageReferenceArgs,
                 {
                     sources.Add(UriUtility.GetAbsolutePath(Environment.CurrentDirectory, source));
                 }
-
                 originalPackageSpec.RestoreMetadata.Sources =
                                     sources.Where(ns => !string.IsNullOrEmpty(ns))
-                                    .Select(ns => new PackageSource(ns))
-                                    .ToList();
+                                    .Select(ns => new PackageSource(ns)) as List<PackageSource>;
```

Comment:
This will simply run ToList() under the hood, and this change would make that less obvious.

Reply:
Removing the ToList() will probably reduce performance, because if `originalPackageSpec.RestoreMetadata.Sources` ends up being enumerated more than once, then the `.Where` and `Select` (which does a `new PackageSource`) will be re-run each time. And the memory allocations will therefore be higher than just doing `ToList()` once.

Unfortunately "avoid ToList()" is an overly simplistic goal which will reduce performance in a bunch of scenarios, not improve it.